subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/WatchableHierarchies.java

```diff
@@ -93,6 +110,29 @@ public SnapshotHierarchy removeWatchedHierarchiesOverLimit(SnapshotHierarchy roo
         return result;
     }
 
+    private SnapshotHierarchy removeUnwatchableFileSystems(SnapshotHierarchy root, Invalidator invalidator) {
+        SnapshotHierarchy invalidatedRoot = watchableFileSystemDetector.detectUnsupportedFileSystems()
+            .reduce(
+                root,
+                (updatedRoot, fileSystem) -> removeUnwatchableFileSystem(updatedRoot, fileSystem, invalidator),
+                nonCombining()
+            );
+        if (invalidatedRoot != root) {
+            LOGGER.info("Some of the file system contents retained in the virtual file system are on file systems that Gradle doesn't support watching. " +
+                "The relevant state was discarded to ensure changes to these locations are properly detected. " +
+                "You can override this by enforcing file system watching.");
```

Comment:
```suggestion
                "You can override this by explicitly enabling file system watching.");
```

I don't think users know what `enforcing` means in this context.